addons/website_blog/static/src/js/website_blog.js

```diff
@@ -12,7 +12,32 @@ publicWidget.registry.websiteBlog = publicWidget.Widget.extend({
         'click #o_wblog_post_content_jump': '_onContentAnchorClick',
         'click .o_twitter, .o_facebook, .o_linkedin, .o_google, .o_twitter_complete, .o_facebook_complete, .o_linkedin_complete, .o_google_complete': '_onShareArticle',
     },
-
+    /**
+     * @override
+     */
+    willStart: function () {
+        const content = this.el.querySelector('.o_wblog_post_content_field');
+        this.el.querySelectorAll(['.o_container_as_first', '.o_container_as_last']).forEach(container => {
+            const which = container.classList.contains('o_container_as_first') ? 'first' : 'last';
+            const source = content.querySelector([
+                `section:${which}-child.o_container_small`,
+                `section:${which}-child.container`,
+                `section:${which}-child.container-fluid`,
+                `section:${which}-child .o_container_small`,
+                `section:${which}-child .container`,
+                `section:${which}-child .container-fluid`,
+            ]);
+            if (source && source.classList.contains('container')) {
+                container.classList.add('container');
+            } else if (source && source.classList.contains('container-fluid')) {
+                container.classList.add('container-fluid');
+            } else {
+                container.classList.add('o_container_small');
+            }
+            container.classList.remove(`o_container_as_${which}`);
+        });
```

Comment:
Mmmh I guess that's a possibility to initialize it via JS on each page load (what I had in mind what to change it via a view xpath depending on what was saved, but probably ok like this too. Should be done on start instead of willStart in the normal public widget's logic though... :)